doc/source/architecture.rst

```diff
@@ -0,0 +1,213 @@
+.. _ref_architecture:
+
+PyMechanical architecture
+=========================
+
+PyMechanical provides a Python interface to Ansys Mechanical. Mechanical is a
+polyglot [#f1]_ desktop application whose graphical user interface (GUI) runs
+on either the Windows or Linux operating system. Mechanical's APIs are
+implemented in C# using .NET Framework 4.x. They are exposed to both C# and
+two implementations [#f2]_ of Python, IronPython [#f3]_ and CPython.
+
+Within Mechanical, Python scripting enables you to automate repetitive GUI
+actions. This is not unlike other applications developed either by Ansys or
+other software companies. Python scripting leverages the Mechanical API.
+
+Mechanical, like some other applications, is customizable. Using the same
+Mechanical API that you use for scripting, you can implement extensions that
+add to the capabilities of Mechanical. For example, buttons can be added to the
+GUI and custom objects can be added to the data model. Even third-party or
+in-house solvers can be integrated into Mechanical and can take advantage of
+the powerful meshing, generic CAD reader, and the intuitive pre- and post-
+processing experience of Mechanical.
+
+Before discussing how Mechanical's API is implemented, a design pattern known
+as the "Command Pattern" is discussed. This is an abstract pattern that can be
+used in any object oriented system in many programming languages. A high level
+explanation of the pattern (using the Java programming language) can be found
+here: https://howtodoinjava.com/design-patterns/behavioral/command-pattern/.
```

Comment:
```suggestion
Before discussing how Mechanical's API is implemented, the *command pattern* is
described. You can use this behavior design pattern to abstract business logic into
discrete actions in any object-oriented system in many programming languages. For
a high-level explanation for using the command pattern in Java, see `Command Design Pattern
 <https://howtodoinjava.com/design-patterns/behavioral/command-pattern/> in the
 *HowToDoInJava* newsletter.
```
I basically incorporated Sean's suggested changes. I also changed the "see here"--which the Goggle dev doc style guide and basically all style guides says is a no-no. You always want to use descriptive link text that explains to the user where the link will take them.